ember-app/app/controllers/milestones/edit.js

```diff
@@ -0,0 +1,63 @@
+import Ember from 'ember';
+
+var MilestonesEditController = Ember.ObjectController.extend({
+  needs: ["milestones"],
+  errors: false,
+  clearErrors: function(){
+    this.set("errors", false);
+  }.observes('model.title', 'model.description', 'model.due_on'),
+  dueDate: function(){
+    return this.get("model.due_on");
+  }.property("model.due_on"),
+  linkedRepos: Ember.computed.alias("controllers.milestones.model.linkedRepos"),
+  actions: {
+    submit: function() {
+      var controller = this;
+      this.set("processing",true);
+      this.editLinkedMilestones();
+
+      this.get("model").saveEdit().then(function(milestone){
+         milestone.originalTitle = controller.get("model.originalTitle");
+         controller.send("milestoneUpdated", milestone);
+         controller.set("processing",false);
+      }).fail(function(){
+         controller.set("processing",false);
+         controller.set("errors", true);
+      });
+    },
+    clearDueDate: function(){
+      this.set("model.due_on", null);
+    }
+  },
+  isCollaboratorBinding: "App.repo.is_collaborator",
+  disabled: function () {
+      return this.get("processing") || !this.get("isValid");
+  }.property("processing","isValid"),
+  isValid: function () {
+    return this.get("model.title");
+  }.property("model.title"),
+
+  editLinkedMilestones: function(){
+    var self = this;
+    var boards = this.get("linkedRepos");
+    var matches = _.filter(boards, board => {
+      return _.find(board.milestones, milestone => {
+        return self.get("model.originalTitle") === milestone.title;
+      });
+    });
+
+    matches.forEach(board => {
+      this.get("model")
+        .saveLinkedEdit(board, this.get("model.originalTitle"))
+        .then(function(milestone){
```

Comment:
jshint did bark thats how i found it.

im still unclear what you're doing when the milestone isn't being used

On Thu, May 14, 2015 at 11:12 AM, discorick <notifications@github.com>
wrote:

> In ember-app/app/controllers/milestones/edit.js
> <https://github.com/huboard/huboard-web/pull/42#discussion_r30334670>:
>
> > +    return this.get("model.title");
> > +  }.property("model.title"),
> > +
> > +  editLinkedMilestones: function(){
> > +    var self = this;
> > +    var boards = this.get("linkedRepos");
> > +    var matches = _.filter(boards, board => {
> > +      return _.find(board.milestones, milestone => {
> > +        return self.get("model.originalTitle") === milestone.title;
> > +      });
> > +    });
> > +
> > +    matches.forEach(board => {
> > +      this.get("model")
> > +        .saveLinkedEdit(board, this.get("model.originalTitle"))
> > +        .then(function(milestone){
>
> Milestone could get removed im surprised JShint didn't bark at me. This
> block is as follows:
>
> Each match is a linked Board, We save the linked edit to the milestone of
> the linked board, then add the new edited milestone to that board's
> milestones list. This will trigger all the appropriate combinedMilestones
> bindings
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/huboard/huboard-web/pull/42/files#r30334670>.
>